presto-tests/src/test/java/io/prestosql/tests/TestWorkProcessorPipelineQueries.java

```diff
@@ -32,6 +32,13 @@ protected TestWorkProcessorPipelineQueries()
     @Test
     public void testTopN()
     {
-        assertQuery("select * from orders order by totalprice limit 10");
+        assertQuery("SELECT * FROM orders ORDER BY totalprice LIMIT 10");
+    }
+
+    @Test
+    public void testSemiJoin()
+    {
+        assertQuery("SELECT AVG(extendedprice) FROM lineitem where partkey IN (SELECT partkey FROM partsupp WHERE supplycost < 495)");
```

Comment:
better query is `SELECT avg(extendedprice) FROM lineitem where orderkey IN (SELECT orderkey FROM orders WHERE orderdate < DATE '1994-01-01');`
It does filter better so it's better example